### PR TITLE
Add ReScript Example

### PR DESCRIPTION
**Pull request by gscarv13**
### This PR contains:
- Add ReScript example component to the `spec/dummy/`
- Boilerplate code provided by `ljuster` at shakacode/react_on_rails#1215

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1400)

### CODE REVIEW

File: spec/dummy/.gitignore [hunk 1]
```diff
@@ -21,3 +21,9 @@
 /public/webpack/*
 
 /spec/examples.txt
+
+#Ignore ReScript build files
+/.merlin
+/lib/bs/
+/.bsb.lock
+*.bs.js
```
Comment (justin808) [inline spec/dummy/.gitignore]: Please note these chars

**Review by justin808 — CHANGES_REQUESTED** [review]
> Reviewed 11 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1400)* status: 11 of 12 files reviewed, 6 unresolved discussions (waiting on @gscarv13 and @justin808)

___
*[spec/dummy/package.json, line 63 at r1](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-Mnc-yhi8Fbii_oWK3oo:-MnbzRYxCOnoa9WyED0k:b-rws4ax) ([raw file](https://github.com/shakacode/react_on_rails/blob/46db32d7b26c2ab9d326ee0ee301e85cb9980469/spec/dummy/package.json#L63)):*
> ```JSON
>     "build:clean": "rm -rf public/webpack || true",
>     "re:build": "rescript build",
>     "re:start": "rescript build -w"
> ```

1. remove these two right here
2. update build:XYZ with the `rescript build`
3. Add "build:production" and set ReactOnRails.configuration.build_production_command
___
*[spec/dummy/client/app/packs/client-bundle.js, line 14 at r1](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-Mnbzz8E1xgSMjx0khvs:-Mnbzz8FDQCHgorj_eCh:b-cyzhfu) ([raw file](https://github.com/shakacode/react_on_rails/blob/46db32d7b26c2ab9d326ee0ee301e85cb9980469/spec/dummy/client/app/packs/client-bundle.js#L14)):*
> ```JavaScript
> import PureComponentWrappedInFunction from '../components/PureComponentWrappedInFunction';
> 
> import { HelloWorldReScript } from './rescript-components';
> ```

@pulkitkkr @alexfedoseev is this the right way to integrate?
___
*[spec/dummy/client/app/packs/rescript-components.js, line 8 at r1](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-Mnbzujt1bi1X2ONQyjj:-Mnbzujt1bi1X2ONQyjk:b-cyzhfu) ([raw file](https://github.com/shakacode/react_on_rails/blob/46db32d7b26c2ab9d326ee0ee301e85cb9980469/spec/dummy/client/app/packs/rescript-components.js#L8)):*
> ```JavaScript
> 
> export {
>   HelloWorldReScript,
> ```

@pulkitkkr @alexfedoseev is this the right way to integrate?
___
*[spec/dummy/lib/tasks/before_assets_precompile.rake, line 3 at r1](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-Mnc3x4r3Ha1yB0Tad4q:-Mnc3x4r3Ha1yB0Tad4r:b-twebmc) ([raw file](https://github.com/shakacode/react_on_rails/blob/46db32d7b26c2ab9d326ee0ee301e85cb9980469/spec/dummy/lib/tasks/before_assets_precompile.rake#L3)):*
> ```Ruby
> # frozen_string_literal: true
> 
> task :before_assets_precompile do
> ```

Once you merge https://github.com/shakacode/react_on_rails/pull/1402 then we can remove this whole file.
___
*[spec/dummy/lib/tasks/before_assets_precompile.rake, line 5 at r1](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-Mnc-e0109xNfS_ejrHx:-Mnc-e0109xNfS_ejrHy:b-f62uup) ([raw file](https://github.com/shakacode/react_on_rails/blob/46db32d7b26c2ab9d326ee0ee301e85cb9980469/spec/dummy/lib/tasks/before_assets_precompile.rake#L5)):*
> ```Ruby
> task :before_assets_precompile do
>   # clean and build rescript files
>   system("yarn rescript clean && yarn rescript build")
> ```

Use `sh`

https://www.rubydoc.info/gems/rake/FileUtils:sh
https://newbedev.com/how-to-execute-commands-within-rake-tasks

throw if command does not run cleanly.

and don't put script here, put script in package.json

`yarn build:

**Review by justin808 — COMMENTED** [review]
Need to add test to https://github.com/shakacode/react_on_rails/blob/master/spec/dummy/spec/system/integration_spec.rb

> Reviewed 1 of 12 files at r1.
> *[Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-:-Mnc4LxL-d3xL_Hs4oxk:bf3ja6e)* status: all files reviewed, 6 unresolved discussions (waiting on @gscarv13)

File: spec/dummy/.gitignore [hunk 2]
```diff
@@ -21,3 +21,9 @@
 /public/webpack/*
 
 /spec/examples.txt
+
+#Ignore ReScript build files
+/.merlin
+/lib/bs/
```
Comment (alex35mil) [inline spec/dummy/.gitignore:33]: The whole `lib` should be ignored.

File: spec/dummy/.prettierignore [hunk 3]
```diff
@@ -1,2 +1,4 @@
 node_modules
 public/webpack/
+*.bs.js
+lib/bs/**
```
Comment (alex35mil) [inline spec/dummy/.prettierignore]: Same.

File: spec/dummy/bsconfig.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    {
+      "dir": "client/app/components",
+      "subdirs": true
+    }
+  ],
+  "reason": {
+    "react-jsx": 3
+  },
+  "bsc-flags": [
+    "-bs-super-errors"
```
Comment (alex35mil) [inline spec/dummy/bsconfig.json]: It's not relevant anymore.

File: spec/dummy/bsconfig.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  ],
+  "suffix": ".bs.js",
+  "namespace": true,
```
Comment (alex35mil) [inline spec/dummy/bsconfig.json]: Useless in the context of the apps.

File: spec/dummy/bsconfig.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  "package-specs": [
+    {
+      "module": "commonjs",
```
Comment (alex35mil) [inline spec/dummy/bsconfig.json]: Is there a reason not to use `es6`?

File: spec/dummy/client/app/packs/rescript-components.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+// This file is used to import your compiled rescript files
+// As per the current configuration, all files are compiled to .bs.js
+// and they are generated on the same directory as the .res file
+
+import { make as HelloWorldReScript } from '../components/HelloWorldReScript.bs';
```
Comment (alex35mil) [inline spec/dummy/client/app/packs/rescript-components.js]: FYI If you export a component for JS, you can call it `default` and import it directly without this `make as` dance.

File: spec/dummy/package.json [hunk 8]
```diff
@@ -51,11 +54,13 @@
...
-    "build:dev": "rm -rf public/webpack/develoment && RAILS_ENV=development NODE_ENV=development bin/webpack --watch",
-    "build:dev:server": "rm -rf public/webpack/develoment && RAILS_ENV=development NODE_ENV=development bin/webpack --watch",
-    "build:dev:watch": "RAILS_ENV=development NODE_ENV=development bin/webpack --watch",
-    "build:clean": "rm -rf public/webpack || true"
+    "build:test": "rm -rf public/webpack/test && yarn build:rescript && RAILS_ENV=test && NODE_ENV=test bin/webpack",
+    "build:dev": "rm -rf public/webpack/develoment && yarn build:rescript && RAILS_ENV=development && NODE_ENV=development bin/webpack --watch",
+    "build:dev:server": "rm -rf public/webpack/develoment && yarn build:rescript && RAILS_ENV=development NODE_ENV=development bin/webpack --watch",
+    "build:dev:watch": "rescript build -w && RAILS_ENV=development NODE_ENV=development bin/webpack --watch",
+    "build:clean": "yarn build:rescript && rm -rf public/webpack || true",
+    "build:rescript": "rescript clean && rescript build",
```
Comment (alex35mil) [inline spec/dummy/package.json]: You probably want `-with-deps` flag here.

File: spec/dummy/package.json [hunk 9]
```diff
@@ -9,7 +9,9 @@
   "dependencies": {
     "@babel/preset-react": "^7.10.4",
     "@rails/webpacker": "^5.2.1",
+    "@rescript/react": "^0.10.3",
     "babel-plugin-module-resolver": "^4.0.0",
+    "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
```
Comment (alex35mil) [inline spec/dummy/package.json]: Is this relevant to ReScript, or is it just unrelated improvement?

File: spec/dummy/.gitignore (hunk 2 above)
Comment (justin808) [reply spec/dummy/.gitignore:33]: @gscarv13 BS files should not be placed in the /lib directory. 

@alexfedoseev what is the bs directory? The lib directory is used by rail, so it should not be ignored.

File: spec/dummy/package.json (hunk 9 above)
Comment (justin808) [reply spec/dummy/package.json]: remove-prop-types is something to save space on production builds.

@gscarv13 why is this being added for rescript?

File: spec/dummy/spec/system/integration_spec.rb [hunk 10]
```diff
@@ -115,6 +115,14 @@ def finished_all_ajax_requests?
   it { is_expected.to have_text "This is a Pure Component!" }
 end
 
+describe "Pages/Hello World ReScript Component", :js do
+  subject { page }
+
+  before { visit "/client_side_rescript_hello_world" }
+
+  it { is_expected.to have_text "Hello from ReScript" }
```
Comment (justin808) [inline spec/dummy/spec/system/integration_spec.rb]: test changing the text in the input field and observing the output

**Review by justin808 — CHANGES_REQUESTED** [review]

File: spec/dummy/package.json (hunk 9 above)
Comment (gscarv13) [reply spec/dummy/package.json]: I was building the app for the tests and that was being required. I'll check if it is still relevant @justin808

File: spec/dummy/client/app/packs/rescript-components.js (hunk 7 above)
Comment (gscarv13) [reply spec/dummy/client/app/packs/rescript-components.js]: Thank You @alexfedoseev 

What about having this file import all `.bs.js` files, is it a good idea?

File: spec/dummy/bsconfig.json (hunk 6 above)
Comment (gscarv13) [reply spec/dummy/bsconfig.json]: I'll update it

File: spec/dummy/.gitignore (hunk 2 above)
Comment (gscarv13) [reply spec/dummy/.gitignore:33]: @justin808 The ReScript compiler put the `bs/` inside `lib/`. Looking at the documentation I couldn't find a way to change it

**Review by justin808 — CHANGES_REQUESTED** [review]
> Reviewed 3 of 6 files at r2, 2 of 3 files at r5, 1 of 3 files at r6, 6 of 6 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1400)* status: 16 of 17 files reviewed, 13 unresolved discussions (waiting on @alexfedoseev, @gscarv13, and @justin808)

___
*[spec/dummy/.gitignore, line 27 at r5](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MnbkJYF4B3udaKpsosi-r5-27:-MouJGPp3Oec-uaAW2NB:b-ic17tg) ([raw file](https://github.com/shakacode/react_on_rails/blob/b6179b392bf11cd5c564bdcf611a93dc8d94abf8/spec/dummy/.gitignore#L27)):*
<details><summary><i>Previously, gscarv13 (Gustavo Carvalho) wrote…</i></summary><blockquote>

@justin808 The ReScript compiler put the `bs/` inside `lib/`. Looking at the documentation I couldn't find a way to change it
</blockquote></details>

Hey @alexfedoseev it's very common to have a /lib directory as part of Rails apps.


@gscarv13 Can ReScript be configured to have the folder called /rescript-lib ?
___
*[spec/dummy/package.json, line 14 at r5](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MnbkJYF4B3udaKpsosn-r5-14:-MouL09d4ivmIg_EDLCM:b-aq6waf) ([raw file](https://github.com/shakacode/react_on_rails/blob/b6179b392bf11cd5c564bdcf611a93dc8d94abf8/spec/dummy/package.json#L14)):*
<details><summary><i>Previously, gscarv13 (Gustavo Carvalho) wrote…</i></summary><blockquote>

I was building the app for the tests and that was being required. I'll check if it is still relevant @justin808 
</blockquote></details>

this is for production builds to remove prop types

so yes, good

and not required in this PR.

separate PR would be good, definitely the update to rails/webpacker
___
*[spec/dummy/client/app/components/HelloWorldReScript.res, line 4 at r7](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MouJl8P72sRGXyIlCRn:-MouJl8P72sRGXyIlCRo:bghju05) ([raw file](https://github.com/shakacode/react_on_rails/blob/f69e21ac509fb54ffb443aab47a73e215ea3e09e/spec/dummy/client/app/components/HelloWorldReScript.res#L4)):*
> ```XML
> // is the same hash passed to react_component method on the view. Log the Props in the compiled JS
> // and make sure the props passed to the component below match the shape of the prop passed on the view.
> ```

Comment on why named `default`
___
*[spec/dummy/client/app/packs/rescript-components.js, line 5 at r7](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MouJw2MC46bV7SE87c6:-MouJw2N0l3tlsn9yrK_:b-h6cjqu) ([raw file](https://github.com/shakacode/react_on_rails/blob/f69e21ac509fb54ffb443aab47a73e215ea3e09e/spec/dummy/client/app/packs/rescript-components.js#L5)):*
> ```JavaScript
> // and they are generated on the same directory as the .res file
> 
> import HelloWorldReScript from '../components/HelloWorldReScript.bs';
> ```

@alexfedoseev should we import the bs file from the JS file like this?

or maybe 

```js
import HelloWorldReScript from '../components/HelloWorldReScript.bs.js';
```

or configure webpack/babel so this works:
```js
import HelloWorldReScript from '../components/HelloWorldReScript';
```
and there is a default to look for the .bs.js file?
___
*[spec/dummy/spec/system/integration_spec.rb, line 123 at r5](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-Mngd59zxqu7eQ-T6rNY-r5-123:-MouKkeO53zM3hVkDYs-:b32w0x) ([raw file](https://github.com/shakacode/react_on_rails/blob/b6179b392bf11cd5c564bdcf611a93dc8d94abf8/spec/dummy/spec/system/integration_spec.rb#L123)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

test changing the text in the input field and observing the output
</blockquote></details>

OK

**Review by mightystrong — COMMENTED** [review]
> Reviewed 3 of 12 files at r1, 2 of 6 files at r2, 2 of 3 files at r5, 3 of 6 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1400)* status: all files reviewed, 14 unresolved discussions (waiting on @alexfedoseev and @gscarv13)

___
*[spec/dummy/.gitignore, line 27 at r5](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MnbkJYF4B3udaKpsosi-r5-27:-Mq_eVpB1uWpAyKSd6X_:bjitt7h) ([raw file](https://github.com/shakacode/react_on_rails/blob/b6179b392bf11cd5c564bdcf611a93dc8d94abf8/spec/dummy/.gitignore#L27)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Hey @alexfedoseev it's very common to have a /lib directory as part of Rails apps.


@gscarv13 Can ReScript be configured to have the folder called /rescript-lib ?
</blockquote></details>

I agree that the lib directory is an odd placement.
___
*[spec/dummy/bsconfig.json, line 13 at r5](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MnbkJYF4B3udaKpsosq-r5-13:-Mq_eoaC0JEu4x-27qGt:bk35s08) ([raw file](https://github.com/shakacode/react_on_rails/blob/b6179b392bf11cd5c564bdcf611a93dc8d94abf8/spec/dummy/bsconfig.json#L13)):*
<details><summary><i>Previously, alexfedoseev (Alex Fedoseev) wrote…</i></summary><blockquote>

It's not relevant anymore.
</blockquote></details>

@alexfedoseev Should we remove it from here too? https://github.com/shakacode/rescript-on-rails-example/blob/main/bsconfig.json
___
*[spec/dummy/app/views/pages/client_side_rescript_hello_world.html.erb, line 4 at r7](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-Mq_gWgbEchnh2vJgr7n:-Mq_gWgc9aR3GQips2-E:b-yinlmv) ([raw file](https://github.com/shakacode/react_on_rails/blob/f69e21ac509fb54ffb443aab47a73e215ea3e09e/spec/dummy/app/views/pages/client_side_rescript_hello_world.html.erb#L4)):*
> ```erb
> <hr/>
> 
> <h1>React Rails Client Side Only Rendering</h1>
> ```

Seems odd to me to mix Rails HTML in a specific view instead of delegating all the front end in the view to ReScript, unless there's a compelling reason. Devs have to bounce between the Rails view and ReScript to decide where to edit the HTML. In my opinion it's cleaner to separate concerns.
___
*[spec/dummy/client/app/packs/rescript-components.js, line 5 at r7](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MouJw2MC46bV7SE87c6:-Mqaj5WoAUhNOO1B7Nnu:b-9qy5xf) ([raw file](https://github.com/shakacode/react_on_rails/blob/f69e21ac509fb54ffb443aab47a73e215ea3e09e/spec/dummy/client/app/packs/rescript-components.js#L5)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

@alexfedoseev should we import the bs file from the JS file like this?

or maybe 

```js
import HelloWorldReScript from '../components/HelloWorldReScript.bs.js';
```

or configure webpack/babel so this works:
```js
import HelloWorldReScript from '../components/HelloWorldReScript';
```
and there is a default to look for the .bs.js file?

</blockquote></details>

@justin808 It would be nice if the `.bs.js` was implied.

**Review by mightystrong — COMMENTED** [review]
> Reviewed 1 of 12 files at r1, 2 of 6 files at r2, 3 of 6 files at r7.
> *[Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1400)* status: all files reviewed, 14 unresolved discussions (waiting on @alexfedoseev and @gscarv13)

**Review by mightystrong — COMMENTED** [review]
> Reviewed 1 of 3 files at r6.
> *[Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1400)* status: all files reviewed, 14 unresolved discussions (waiting on @alexfedoseev and @gscarv13)

**Review by gscarv13 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1400)* status: 9 of 17 files reviewed, 14 unresolved discussions (waiting on @alexfedoseev, @gscarv13, @justin808, and @mightystrong)

___
*[spec/dummy/.gitignore, line 29 at r1](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MnbkJYF4B3udaKpsosi-r1-29:-Ms1wXiVBHwpYnyqQehS:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/46db32d7b26c2ab9d326ee0ee301e85cb9980469/spec/dummy/.gitignore#L29)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Please note these chars
</blockquote></details>

Done.
___
*[spec/dummy/.gitignore, line 27 at r5](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MnbkJYF4B3udaKpsosi-r5-27:-Ms5czL3AqxmO1HHP55J:b-poobru) ([raw file](https://github.com/shakacode/react_on_rails/blob/b6179b392bf11cd5c564bdcf611a93dc8d94abf8/spec/dummy/.gitignore#L27)):*
<details><summary><i>Previously, mightystrong (Michael Price) wrote…</i></summary><blockquote>

I agree that the lib directory is an odd placement.
</blockquote></details>

Sadly there's no way of configuring it in rescript yet https://github.com/rescript-lang/rescript-compiler/issues/1124#issuecomment-877618038
___
*[spec/dummy/bsconfig.json, line 13 at r5](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MnbkJYF4B3udaKpsosq-r5-13:-Ms5dxy54AtO4KklYuK4:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/b6179b392bf11cd5c564bdcf611a93dc8d94abf8/spec/dummy/bsconfig.json#L13)):*
<details><summary><i>Previously, mightystrong (Michael Price) wrote…</i></summary><blockquote>

@alexfedoseev Should we remove it from here too? https://github.com/shakacode/rescript-on-rails-example/blob/main/bsconfig.json
</blockquote></details>

Done.
___
*[spec/dummy/bsconfig.json, line 16 at r5](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MnbkJYF4B3udaKpsosq-r5-16:-Ms5dzCy8_s60p6ngoPK:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/b6179b392bf11cd5c564bdcf611a93dc8d94abf8/spec/dummy/bsconfig.json#L16)):*
<details><summary><i>Previously, alexfedoseev (Alex Fedoseev) wrote…</i></summary><blockquote>

Useless in the context of the apps.
</blockquote></details>

Done.
___
*[spec/dummy/package.json, line 62 at r5](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MnbkJYF4B3udaKpsosn-r5-62:-Ms5eKUp5KoSDSxO04b2:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/b6179b392bf11cd5c564bdcf611a93dc8d94abf8/spec/dummy/package.json#L62)):*
<details><summary><i>Previously, alexfedoseev (Alex Fedoseev) wrote…</i></summary><blockquote>

You probably want `-with-deps` flag here.
</blockquote></details>

Done.
___
*[spec/dummy/app/views/pages/client_side_rescript_hello_world.html.erb, line 4 at r7](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-Mq_gWgbEchnh2vJgr7n:-Ms5ekXwCHhSO_xdIAhK:b-cpsj64) ([raw file](https://github.com/shakacode/react_on_rails/blob/f69e21ac509fb54ffb443aab47a73e215ea3e09e/spec/dummy/app/views/pages/client_side_rescript_hello_world.html.erb#L4)):*
<details><summary><i>Previously, mightystrong (Michael Price) wrote…</i></summary><blockquote>

Seems odd to me to mix Rails HTML in a specific view instead of delegating all the front end in the view to ReScript, unless there's a compelling reason. Devs have to bounce between the Rails view and ReScript to decide where to edit the HTML. In my opinion it's cleaner to separate concerns.
</blockquote></details>

Since this is already used in another view, I'll remove it from this one
___
*[spec/dummy/client/app/components/HelloWorldReScript.res, line 4 at r7](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MouJl8P72sRGXyIlCRn:-Ms5fU7MAvQi-0r5hMdk:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/f69e21ac509fb54ffb443aab47a73e215ea3e09e/spec/dummy/client/app/components/HelloWorldReScript.res#L4)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Comment on why named `default`
</blockquote></details>

Done.
___
*[spec/dummy/client/app/packs/rescript-components.js, line 5 at r7](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MouJw2MC46bV7SE87c6:-Ms5lXHHArjs8cK5fGu_:b-s0b6jj) ([raw file](https://github.com/shakacode/react_on_rails/blob/f69e21ac509fb54ffb443aab47a73e215ea3e09e/spec/dummy/client/app/packs/rescript-components.js#L5)):*
<details><summary><i>Previously, mightystrong (Michael Price) wrote…</i></summary><blockquote>

@justin808 It would be nice if the `.bs.js` was implied.
</blockquote></details>

I find it better to be explicit and import as `.bs.js';` to make clear it's a rescript component
___
*[spec/dummy/lib/tasks/before_assets_precompile.rake, line 3 at r1](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-Mnc3x4r3Ha1yB0Tad4q:-Ms5mfQwBdzxUCSr4OQx:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/46db32d7b26c2ab9d326ee0ee301e85cb9980469/spec/dummy/lib/tasks/before_assets_precompile.rake#L3)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Once you merge https://github.com/shakacode/react_on_rails/pull/1402 then we can remove this whole file.
</blockquote></details>

Done.
___
*[spec/dummy/lib/tasks/before_assets_precompile.rake, line 5 at r1](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-Mnc-e0109xNfS_ejrHx:-Ms5mh0hANyO4Tvns0Nf:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/46db32d7b26c2ab9d326ee0ee301e85cb9980469/spec/dummy/lib/tasks/before_assets_precompile.rake#L5)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Use `sh`

https://www.rubydoc.info/gems/rake/FileUtils:sh
https://newbedev.com/how-to-execute-commands-within-rake-tasks

throw if command does not run cleanly.

and don't put script here, put script in package.json

`yarn build:
</blockquote></details>

Done.

**Review by gscarv13 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1400)* status: 9 of 17 files reviewed, 14 unresolved discussions (waiting on @alexfedoseev, @gscarv13, @justin808, and @mightystrong)

___
*[spec/dummy/package.json, line 14 at r5](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MnbkJYF4B3udaKpsosn-r5-14:-Ms5rPfP0mG12GXzGBic:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/b6179b392bf11cd5c564bdcf611a93dc8d94abf8/spec/dummy/package.json#L14)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

this is for production builds to remove prop types

so yes, good

and not required in this PR.

separate PR would be good, definitely the update to rails/webpacker
</blockquote></details>

Done.

**Review by justin808 — COMMENTED** [review]
Looks ready to merge. So long as CI passes, we can merge.

> Reviewed 2 of 4 files at r8, 6 of 6 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-:-Ms7Pb-M1FxFbzUTC7Ck:bz3i2na)* status: all files reviewed, 9 unresolved discussions (waiting on @alexfedoseev, @gscarv13, and @mightystrong)

___
*[spec/dummy/client/app/packs/rescript-components.js, line 5 at r7](https://reviewable.io/reviews/shakacode/react_on_rails/1400#-MouJw2MC46bV7SE87c6:-Ms7POxz2qsMwhx9cmvu:b2nhksp) ([raw file](https://github.com/shakacode/react_on_rails/blob/f69e21ac509fb54ffb443aab47a73e215ea3e09e/spec/dummy/client/app/packs/rescript-components.js#L5)):*
<details><summary><i>Previously, gscarv13 (Gustavo Carvalho) wrote…</i></summary><blockquote>

I find it better to be explicit and import as `.bs.js';` to make clear it's a rescript component
</blockquote></details>

OK

**Pull request merged**